src/library/scala/concurrent/duration/Duration.scala

```diff
@@ -124,6 +124,8 @@ object Duration {
       Undefined
     else if (nanos > Long.MaxValue || nanos < Long.MinValue)
       throw new IllegalArgumentException("trying to construct too large duration with " + nanos + "ns")
+    else if (nanos < 0)
+      fromNanos((nanos - 0.5).toLong)
```

Comment:
Also: IIUC, scala/bug#10320 only affects `Double` which uses all of the available 52 bits of mantissa. More bugs might lurk elsewhere. I'm no IEEE-754 expert, but let's maybe not try to reimplement `round`, at least not without significantly more coverage. Especially not when we're taking the effort to parse a String anyway.